runtime/registry_test.go

```diff
@@ -0,0 +1,279 @@
+			// load all caches
+			svc, err := rt.catalogCache.get(ctx, rt, inst.ID)
+			require.NoError(t, err)
+
+			// ingest some data
+			require.NoError(t, svc.Olap.Exec(ctx, &drivers.Statement{Query: "CREATE TABLE data(id INTEGER, name VARCHAR)"}))
+			require.NoError(t, svc.Olap.Exec(ctx, &drivers.Statement{Query: "INSERT INTO data VALUES (1, 'Mark'), (2, 'Hannes')"}))
+			require.NoError(t, svc.Catalog.CreateEntry(ctx, "default", &drivers.CatalogEntry{
+				Name: "data",
+				Type: drivers.ObjectTypeTable,
+				Object: &runtimev1.Table{
+					Name:    "data",
+					Managed: true,
+				},
+			}))
+
+			// delete instance
+			err = rt.DeleteInstance(ctx, tt.instanceID, tt.dropDB)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("Runtime.DeleteInstance() error = %v, wantErr %v", err, tt.wantErr)
+			}
+			if err != nil {
+				return
+			}
+
+			// verify db is correctly cleared
+			_, err = rt.FindInstance(ctx, inst.ID)
+			require.Error(t, err)
+
+			// verify older olap connection is closed and cache updated
+			require.False(t, rt.connCache.cache.Contains(inst.ID+inst.OLAPDriver+inst.OLAPDSN))
+			require.False(t, rt.connCache.cache.Contains(inst.ID+inst.RepoDriver+inst.RepoDSN))
+			_, ok := rt.catalogCache.cache[inst.ID]
+			require.False(t, ok)
+			_, err = svc.Olap.Execute(context.Background(), &drivers.Statement{Query: "SELECT COUNT(*) FROM rill.migration_version"})
+			require.True(t, err != nil)
+
+			if tt.dropDB {
+				// get a new connection to verify db schema and ingested data is dropped
+				olap, err := drivers.Open(inst.OLAPDriver, inst.OLAPDSN, rt.logger)
+				require.NoError(t, err)
+				olapStore, _ := olap.OLAPStore()
+
+				// verify rillschema is dropped
+				rows, err := olapStore.Execute(ctx, &drivers.Statement{Query: "SELECT count(*) FROM information_schema.schemata WHERE schema_name = 'rill'"})
+				require.NoError(t, err)
+				var count int
+				require.True(t, rows.Next())
+				require.NoError(t, rows.Scan(&count))
+				require.Equal(t, 0, count)
+				require.NoError(t, rows.Close())
+
+				// verify ingested data is dropped
+				rows, err = olapStore.Execute(ctx, &drivers.Statement{Query: "SELECT count(*) FROM information_schema.tables WHERE table_name = 'data'"})
+				require.NoError(t, err)
+				require.True(t, rows.Next())
+				require.NoError(t, rows.Scan(&count))
+				require.Equal(t, 0, count)
+				require.NoError(t, rows.Close())
+
+			}
+		})
+	}
+}
+
+// New returns a runtime configured for use in tests.
+func NewTestRunTime(t *testing.T) *Runtime {
+	opts := &Options{
+		ConnectionCacheSize: 100,
+		MetastoreDriver:     "sqlite",
+		// Setting a test-specific name ensures a unique connection when "cache=shared" is enabled.
+		// "cache=shared" is needed to prevent threading problems.
+		MetastoreDSN:   fmt.Sprintf("file:%s?mode=memory&cache=shared", t.Name()),
+		QueryCacheSize: 10000,
+	}
+	rt, err := New(opts, nil)
+	require.NoError(t, err)
+
+	return rt
+}
```

Comment:
Look in `catalog_test` – there's a `testruntime` package, can't that be used?

Reply:
I actually copied it from there only since `testruntime` package can't be used here.

Reply:
Makes sense